@here/harp-mapview/lib/geometry/PhasedTileGeometryLoader.ts

```diff
@@ -0,0 +1,420 @@
+     *      disabled.
+     * @returns {boolean} `true` if actual geometry has been created.
+     */
+    updateCompletely(
+        enabledKinds: GeometryKindSet | undefined,
+        disabledKinds: GeometryKindSet | undefined
+    ): boolean {
+        return this.update(enabledKinds, disabledKinds, true);
+    }
+
+    /**
+     * Update the tile to the specified phase (index). All intermediate phases between the current
+     * and the specified phase will be processed.
+     *
+     * @param toPhase A value between 0 and `numberOfPhases`.
+     * @param {(GeometryKindSet | undefined)} enabledKinds The [[GeometryKind]]s that should be
+     *      enabled.
+     * @param {(GeometryKindSet | undefined)} disabledKinds The [[GeometryKind]]s that should be
+     *      disabled.
+     * @returns {boolean} `true` if actual geometry has been created.
+     */
+    updateToPhase(
+        toPhase: number,
+        enabledKinds: GeometryKindSet | undefined,
+        disabledKinds: GeometryKindSet | undefined
+    ): boolean {
+        let didUpdate = false;
+        toPhase = Math.min(toPhase, this.numberOfPhases);
+        while (this.currentPhase < toPhase) {
+            didUpdate = this.update(enabledKinds, disabledKinds);
+            if (!didUpdate) {
+                break;
+            }
+        }
+        return didUpdate;
+    }
+
+    /**
+     * Create all geometries in the current phase, or ignore the phases and create all remaining
+     * geometries if `doFullUpdate` is `true`.
+     *
+     * @param {(GeometryKindSet | undefined)} enabledKinds The [[GeometryKind]]s that should be
+     *      enabled.
+     * @param {(GeometryKindSet | undefined)} disabledKinds The [[GeometryKind]]s that should be
+     *      disabled.
+     * @param doFullUpdate If a value of `true` is specified, the current phase is ignored and all
+     *      remaining geometries are created.
+     * @returns {boolean} `true` if actual geometry has been created.
+     */
+    update(
+        enabledKinds: GeometryKindSet | undefined,
+        disabledKinds: GeometryKindSet | undefined,
+        doFullUpdate: boolean = false
+    ): boolean {
+        const tile = this.tile;
+        const loadPhaseDefinitions = this.m_loadPhaseDefinitions;
+
+        if (!tile.dataSource.cacheable) {
+            this.m_currentPhaseIndex = loadPhaseDefinitions.length;
+            return false;
+        }
+
+        let decodedTile = this.m_decodedTile;
+        const currentPhase = this.currentPhase;
+
+        // First time this tile is handled:
+        if (decodedTile === undefined && tile.decodedTile !== undefined) {
+            decodedTile = this.setDecodedTile(tile.decodedTile);
+            this.processTechniques(enabledKinds, disabledKinds);
+        }
+
+        if (decodedTile === undefined || currentPhase >= this.numberOfPhases) {
+            return false;
+        }
+
+        const geometryCreator = new TileGeometryCreator();
```

Comment:
Can we use a singleton instance of TileGeometryCreator instead? I took a quick look and it appears that class has no instance data.